DigitalLearningSolutions.Data.Migrations/202011101344_CompetencyFrameworkDBChanges.cs

```diff
@@ -0,0 +1,172 @@
+﻿namespace DigitalLearningSolutions.Data.Migrations
+{
+    using FluentMigrator;
+    using FluentMigrator.SqlServer;
+    [Migration(202011101344)]
+    public class CompetencyFrameworkDBChanges : Migration
+    {
+        public override void Up()
+        {
+            //DLSV2-102 Add IsFrameworkDeveloper bit field to AdminUsers:
+            Alter.Table("AdminUsers")
+                .AddColumn("IsFrameworkDeveloper").AsBoolean().NotNullable().WithDefaultValue(false);
+            //DLSV2-41 implied publish status table:
+            Create.Table("PublishStatus")
+                .WithColumn("ID").AsInt32().NotNullable().PrimaryKey().Identity()
+                .WithColumn("Status").AsString(255).NotNullable();
+            Insert.IntoTable("PublishStatus").Row(new { Status = "Draft" });
+            Insert.IntoTable("PublishStatus").Row(new { Status = "In Review" });
+            Insert.IntoTable("PublishStatus").Row(new { Status = "Published" });
+            //DLSV2-41 new frameworks table and DLSV2-35 with FrameworkConfig field:
+            Create.Table("Frameworks")
```

Comment:
Unrelated to layout and error pages, but should frameworks really be a separate table to SelfAssessments? Aren't they really representing the same thing? 🙂

Reply:
Again - these were on another branch and somehow got dragged into the pull request, sorry :-(
Frameworks aren't the same as Self Assessments now that I understand it better. Self Assessments let a manager pull together competencies from one or more frameworks to present to a user. The frameworks are created and managed by a new framework developer role and I'm being asked to start work on the interface for that based on the LDA project outputs.

Reply:
Ah okay 🙂